From 8f213aad4b33f481e5bc50bb1b96a7c33dba595c Mon Sep 17 00:00:00 2001
From: Al Viro <viro@zeniv.linux.org.uk>
Date: Wed, 18 Jul 2012 17:32:50 +0400
Subject: [PATCH 069/132] unobfuscate follow_up() a bit

really convoluted test in there has grown up during struct mount
introduction; what it checks is that we'd reached the root of
mount tree.

Change-Id: Ie6a9cfde6d669e3b59a58bca8f71204d41bbc9bd
---
 fs/namei.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/namei.c b/fs/namei.c
index a1e55e51557..5d66edb9415 100644
--- a/fs/namei.c
+++ b/fs/namei.c
@@ -719,7 +719,7 @@ int follow_up(struct path *path)
 
 	br_read_lock(&vfsmount_lock);
 	parent = mnt->mnt_parent;
-	if (&parent->mnt == path->mnt) {
+	if (parent == mnt) {
 		br_read_unlock(&vfsmount_lock);
 		return 0;
 	}
-- 
2.17.1

